src/frontend/src/common/components/library/data_subview/components/createTreeModal/index.tsx

```diff
@@ -40,33 +43,35 @@ interface Props {
   failedSamples: any[];
   open: boolean;
   onClose: () => void;
-  setCreateTreeFailed: (hasFailed: boolean) => void;
+  handleCreateTreeFailed: () => void;
+  handleSetCreateTreeStarted: () => void;
 }
 
 export const CreateTreeModal = ({
   sampleIds,
   failedSamples,
   open,
   onClose,
-  setCreateTreeFailed,
+  handleCreateTreeFailed,
+  handleSetCreateTreeStarted,
 }: Props): JSX.Element => {
   const [treeName, setTreeName] = useState<string>("");
   const [isTreeNameTooLong, setTreeNameTooLong] = useState<boolean>(false);
   const [isTreeBuildDisabled, setTreeBuildDisabled] = useState<boolean>(false);
-  const [treeType, setTreeType] = useState<string>("");
-  const [isContextual, setContextual] = useState<boolean>(true);
-  const [isLocal, setLocal] = useState<boolean>(false);
+  const [treeType, setTreeType] = useState<string>("TARGETED");
```

Comment:
Also, this seems like a candidate for more fixed Types. Not just `string`, but a union of literal string types
`"TARGETED" | "NON_CONTEXTUALIZED" | ...etc...`
That way we have a clear spot in the code of what the allowed tree types are and we'll get TS type checking if there's ever a typo.

Could also go farther and define a fixed object of the allowed values, make a `TREE_TYPES` string enum (although I don't think I like enums generally [that said, I've only read them in the TS handbook], I think string enums are useful for locking in a fixed set of options that need to be referenced in various places -- I did similar stuff a lot in past work with constant objects when we needed fixed strings for complicated switch-case stuff).

Reply:
Yeah i was thinking about making an enum for this but i actually like the idea of creating a custom type instead. I'll add that